lib/LuxLib/src/impl/matmul.jl

```diff
@@ -244,105 +244,15 @@ function CRC.rrule(
     return matmuladd(A, B, bias), ∇matmuladd
 end
 
-# COV_EXCL_START
-
 # EnzymeRules
-function EnzymeRules.augmented_primal(
-    cfg,
-    ::EnzymeCore.Const{typeof(matmuladd!)},
-    ::Type{EnzymeCore.Const{Nothing}},
-    C::EnzymeCore.Annotation{<:AbstractMatrix},
-    opmode::EnzymeCore.Const{<:AbstractInternalArrayOpMode},
-    A::EnzymeCore.Annotation{<:AbstractMatrix},
-    B::EnzymeCore.Annotation{<:AbstractMatrix},
-    bias::EnzymeCore.Annotation{<:AbstractVector},
+EnzymeRules.@easy_rule(
+    matmul(A::AbstractMatrix, B::AbstractMatrix), (matmul(Ω, B'), matmul(A', Ω)),
 )
-    A_cache =
-        if EnzymeRules.overwritten(cfg)[4] &&
-            !(B isa EnzymeCore.Const) &&
-            !(C isa EnzymeCore.Const)
-            copy(A.val)
-        else
-            nothing
-        end
-    B_cache =
-        if EnzymeRules.overwritten(cfg)[5] &&
-            !(A isa EnzymeCore.Const) &&
-            !(C isa EnzymeCore.Const)
-            copy(B.val)
-        else
-            nothing
-        end
-
-    if !(C isa EnzymeCore.DuplicatedNoNeed || C isa EnzymeCore.BatchDuplicatedNoNeed)
-        matmuladd!(C.val, opmode.val, A.val, B.val, bias.val)
-    end
-
-    return EnzymeRules.AugmentedReturn(nothing, nothing, (A_cache, B_cache))
-end
 
-function EnzymeRules.reverse(
-    cfg,
```

Comment:
the in place manual rules are still going to be more efficient than the easy rule